client/src/main/java/org/apache/crail/conf/CrailConfiguration.java

```diff
@@ -33,9 +33,32 @@
 	private static final Logger LOG = CrailUtils.getLogger();
 	private ConcurrentHashMap<String, String> conf;
 
-	public CrailConfiguration() throws IOException {
+	private CrailConfiguration() {
 		conf = new ConcurrentHashMap<>();
-		Properties properties = loadProperties("crail-site.conf");
+	}
+
+	public static CrailConfiguration createEmptyConfiguration() {
+		return new CrailConfiguration();
+	}
+
+	public static CrailConfiguration createConfigurationFromFile() throws IOException {
+		CrailConfiguration cconf = createEmptyConfiguration();
+		String base = System.getenv("CRAIL_HOME");
+		if (base == null || base.isEmpty()) {
+			throw new IllegalArgumentException("CRAIL_HOME environment variable is not set or empty");
+		}
+		cconf.loadConfigurationFromFile(base + "/conf/crail-site.conf");
```

Comment:
I recommend using File.separator instead of '/', makes it more portable.

Reply:
Good idea!

Done.